src/components/CodeView/index.tsx

```diff
@@ -161,4 +177,16 @@ export class CodeViewBase extends React.Component<Props> {
   }
 }
 
-export default withRouter(CodeViewBase);
+const mapStateToProps = (
+  state: ApplicationState,
+  ownProps: RouteComponentProps,
+): PropsFromState => {
+  const { location } = ownProps;
+  const messageUid = queryString.parse(location.search)[messageUidQueryParam];
```

Comment:
Why does this need to happen in `mapStateToProps`? Can't it just happen in `render()`? Introducing `connect()` just for this seems wasteful since it registers update listeners to the Redux store.

Reply:
This is a good point. I've eliminated the issue by moving the logic into `LinterMessage` as you suggested in another comment.